### PR TITLE
*: run 'go test' with CGO_ENABLED=0

### PR DESCRIPTION
**Pull request by ti-mo**
This PR makes the test suite runnable with CGo disabled.

- Avoids gcc dependency on our CI runners and/or test images
- No need for a C cross-compiler, making testing on other arches significantly less painful
- Slight compile time reduction

With 'go clean -testcache ./...' and 'go clean --cache' in between runs:

master: CGO_ENABLED=1 go test ./...  752.56s user 57.52s system 1217% cpu 1:06.56 total
branch: CGO_ENABLED=0 go test ./...  729.65s user 54.95s system 1217% cpu 1:04.43 total

```release-note
Run 'go test' with CGO_ENABLED=0
```

### CODE REVIEW

File: proxylib/proxylib/types.go [hunk 1]
```diff
@@ -86,6 +87,8 @@ func (r FilterResult) Error() string {
 		return "INVALID_ADDRESS"
 	case INVALID_INSTANCE:
 		return "INVALID_INSTANCE"
+	case UNKNOWN_ERROR:
+		return "UNKNOWN_ERROR"
 	}
-	return "UNKNOWN_ERROR"
+	return ""
```
Comment (jrajahalme) [inline proxylib/proxylib/types.go]: I would not change the default case to an empty string. Maybe cast to an int and format as a string?

**Review by jrajahalme — APPROVED** [review]
Assuming the tests pass...

File: proxylib/proxylib/types.go (hunk 1 above)
Comment (ti-mo) [reply proxylib/proxylib/types.go]: Good point, thanks.

**Review by sayboras — APPROVED** [review]
LGTM as long as the tests are ✅

**Review by qmonnet — APPROVED** [review]
Looks all good, thanks a lot!

**Pull request merged**